### ISSUE TITLE
3.3 Crashing on One of My Computers (running from the same Google Drive folder)

### ISSUE THREAD

**Issue by dsnvwlmnt** [PoESkillTree/PoESkillTree#510]
Stack trace seems to indicate you have a hardcoded path... Maybe.

> The error time: 30-May-18 10:24 PM
> Exception: System.UnauthorizedAccessException: Access to the path 'Builds' is denied.
>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.Directory.InternalCreateDirectory(String fullPath, String path, Object dirSecurityObj, Boolean checkHost)
>    at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
>    at System.IO.Directory.CreateDirectory(String path)
>    at POESKillTree.Model.Serialization.AbstractPersistentDataDeserializer.<InitializeAsync>d__22.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Model\Serialization\AbstractPersistentDataDeserializer.cs:line 87
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at POESKillTree.Model.Serialization.PersistentDataSerializationService.PersistentData.<InitializeAsync>d__5.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Model\Serialization\PersistentDataSerializationService.cs:line 119
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
>    at POESKillTree.Views.MainWindow.<Window_Loaded>d__90.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Views\MainWindow.xaml.cs:line 475
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
>    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
>    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by EmmittJ** [comment]
Try running the program as admin? It looks like it is not able to access your builds folder.

> System.UnauthorizedAccessException: Access to the path 'Builds' is denied.

**Comment by dsnvwlmnt** [comment]
That worked, but it seems to have deleted all my builds in the process. I restored the missing files, but the program seems to refuse to point to my same old Builds folder. If i point it to the right one, it ends up still pointing to something inaccessible in C:\Windows\System32.

**Comment by EmmittJ** [comment]
You can change the build path in the options menu. Is it pointed to your build folder?

**Comment by dsnvwlmnt** [comment]
If i point it to the right one, it ends up still pointing to something inaccessible in C:\Windows\System32.

**Comment by EmmittJ** [comment]
Ok, try running not as admin?

**Comment by dsnvwlmnt** [comment]
Program loads but i got:

`An error occurred during a save operation. Details. Access to the path 'C:\WINDOWS\system32\Builds\.buildfolder' is denied`

Then when i tried to change the build folder, it crashed after choosing it with this trace:

```
The error time: 31-May-18 10:04 AM
Exception: System.UnauthorizedAccessException: Access to the path 'C:\WINDOWS\system32\Builds\.buildfolder' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path)
   at POESKillTree.Utils.SerializationUtils.XmlSerializeToFile[T](T obj, String filePath) in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Utils\SerializationUtils.cs:line 56
   at POESKillTree.Model.Serialization.PersistentDataSerializer.SerializeFolder(String path, BuildFolder folder) in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Model\Serialization\PersistentDataSerializer.cs:line 271
   at POESKillTree.Model.Serialization.PersistentDataSerializer.SerializeFolders() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Model\Serialization\PersistentDataSerializer.cs:line 132
   at POESKillTree.Model.Serialization.PersistentDataSerializationService.PersistentData.SaveFolders() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Model\Serialization\PersistentDataSerializationService.cs:line 145
   at POESKillTree.ViewModels.SettingsMenuViewModel.<ChangeBuildsSavePath>d__12.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\ViewModels\SettingsMenuViewModel.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Common.ViewModels.AsyncRelayCommand`1.<Execute>d__2.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSKillTree\Common\ViewModels\RelayCommand.cs:line 132
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by EmmittJ** [comment]
Are you running the portable version or the installed version?

**Comment by dsnvwlmnt** [comment]
Portable

**Comment by EmmittJ** [comment]
Oh, sorry. I missed the Google Drive bit in the title. Is it possible that where Google Drive is mapped is different between the devices?

**Comment by EmmittJ** [comment]
Hmm, it could be that it is breaking when trying to access the path from another system. Could you unzip a clean version from the latest release, setup the builds path then copy your builds over?

**Comment by brather1ng** [comment]
Build paths set through the settings will always be absolute, so that won't help when running it from different devices.

For portable installations, the default path makes the most sense, as it should be relative to the program exe. But for you relative paths somehow seem to be based on C:\Windows\System32, which causes access exceptions when not running as admin. Do you pass any options to the exe or shortcut of the exe?

**Comment by dsnvwlmnt** [comment]
You nailed it, the path on my shortcut was messed up for "Start in".

On the computer that it didn't work (and used to work):
Target: "C:\Users\unsane\Google Drive\Documents\portable\PoESkillTree\PoESkillTree.exe"
Start in: "C:\Users\unsane\Google Drive\My Documents\portable\PoESkillTree"

On the computer that works:
"D:\Google Drive\Documents\portable\PoESkillTree\PoESkillTree.exe"
"D:\Google Drive\Documents\portable\PoESkillTree"

Basically i changed 'My Documents' to 'Documents' on my old computers when i got a new one, but knew there would be some niche problems like this that would be impossible to predict.

Sorry, and thanks!
